cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis/CreateTestnetData.hs

```diff
@@ -338,40 +341,46 @@ runGenesisCreateTestNetDataCmd
 
     addDRepsToConwayGenesis
       :: [VerificationKey DRepKey]
+      -- \^ The credential of the DReps
       -> [VerificationKey StakeKey]
+      -- \^ The credentials of those that delegate their votes to the dreps. It happens
+      -- to only be stake keys now, but it could be more general in the future.
       -> L.ConwayGenesis L.StandardCrypto
+      -- \^ The genesis data to amend
       -> L.ConwayGenesis L.StandardCrypto
     addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
-      conwayGenesis
-        { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
-        , L.cgInitialDReps = initialDReps (L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis) dRepKeys
-        }
+      conwayGenesis{L.cgDelegs, L.cgInitialDReps}
      where
-      delegs
-        :: [(VerificationKey StakeKey, VerificationKey DRepKey)]
-        -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      -- The credential, to the drep it delegates to
+      delegs :: [(L.Credential L.Staking L.StandardCrypto, VerificationKey DRepKey)]
       delegs =
-        fromList
-          . map
-            ( bimap
-                verificationKeytoStakeCredential
-                (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential)
-            )
+        map
+          (first verificationKeytoStakeCredential)
+          (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+      -- If there are more staking keys than dreps, some dreps don't receive any delegation
+      drepsWithoutDelegation = [drep | drep <- dRepKeys, drep `notElem` map snd delegs]
```

Comment:
Ah, this `notElem` over the list may be an issue, it looks cuadratic, also it could be a `filter` instead

Reply:
Yes, that would be logarithmic. But I think it must be possible to filter based on the position on the list (if the position is smaller or equal than the number of delegators, or something like that), and then it would be linear.

Reply:
I did the following change in an additional commit:

```diff
-      drepsWithoutDelegation = [drep | drep <- dRepKeys, drep `notElem` map snd delegs]
+      drepsWithoutDelegation
+        | length stakingKeys >= length dRepKeys = []
+        | otherwise = drop (length stakingKeys) dRepKeys
```

and this brings execution time back to the original state, while memory usage increases (from 1.95G to 3.3G):

```shell
# Before this PR
→ /usr/bin/time -v cabal run cardano-cli -- latest genesis create-testnet-data --transient-stake-delegators 1000000 --transient-drep-keys 1000000 --out-dir test
	Command being timed: "cabal run cardano-cli -- latest genesis create-testnet-data --transient-stake-delegators 1000000 --transient-drep-keys 1000000 --out-dir test"
	User time (seconds): 40.67
	System time (seconds): 1.04
	Percent of CPU this job got: 100%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:41.63
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 1956096
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 521258
	Voluntary context switches: 4282
	Involuntary context switches: 313
	Swaps: 0
	File system inputs: 0
	File system outputs: 555128
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
# After this PR
→ /usr/bin/time -v cabal run cardano-cli -- latest genesis create-testnet-data --transient-stake-delegators 1000000 --transient-drep-keys 1000000 --out-dir test
	Command being timed: "cabal run cardano-cli -- latest genesis create-testnet-data --transient-stake-delegators 1000000 --transient-drep-keys 1000000 --out-dir test"
	User time (seconds): 42.90
	System time (seconds): 1.53
	Percent of CPU this job got: 100%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:44.37
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 3331840
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 865235
	Voluntary context switches: 4557
	Involuntary context switches: 360
	Swaps: 0
	File system inputs: 8
	File system outputs: 691856
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
```